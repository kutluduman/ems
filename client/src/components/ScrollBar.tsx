import { useEffect } from "react";
import SimpleBarReact from "simplebar-react";
import { alpha, styled } from "@mui/material/styles";
import { Box } from "@mui/material";


const RootStyle = styled("div")({
  flexGrow: 1,
  height: "100%",
  overflow: "hidden",
});

const SimpleBarStyle = styled(SimpleBarReact)(({ theme }) => ({
  maxHeight: "100%",

  "& .simplebar-wrapper": {
    "&.simplebar-placeholder": {
      height: 0,
    },
  },
  "& .simplebar-scrollbar": {
    "&:before": {
      backgroundColor: alpha(theme.palette.grey[600], 0.48),
    },
    "&.simplebar-visible:before": {
      opacity: 1,
    },
  },
  "& .simplebar-track.simplebar-vertical": {
    width: 10,
  },
  "& .simplebar-track.simplebar-horizontal .simplebar-scrollbar": {
    height: 6,
  },
  "& .simplebar-mask": {
    zIndex: "inherit",
  },
}));


interface ScrollbarProps {
  [key: string]: any;
}

const Scrollbar = ({ children, sx, ...other }: ScrollbarProps) => {
  const isMobile =
    /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
      navigator.userAgent
    );

  useEffect(() => {
    const simpleBarPlaceholder = document.querySelector(
      ".simplebar-placeholder"
    ) as HTMLDivElement;
    if (simpleBarPlaceholder) {
      simpleBarPlaceholder.style.height = "0px";
    }
  }, []);

  if (isMobile) {
    return (
      <Box sx={{ overflowX: "auto", ...sx }} {...other}>
        {children}
      </Box>
    );
  }

  return (
    <RootStyle>
      <SimpleBarStyle timeout={500} clickOnTrack={false} sx={sx} {...other}>
        {children}
      </SimpleBarStyle>
    </RootStyle>
  );
}
export default Scrollbar;
