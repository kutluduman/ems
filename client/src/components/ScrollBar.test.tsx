import { create } from "react-test-renderer";

import ScrollBar from "./ScrollBar";

describe("ScrollBar Component", () => {
  test("renders the ScrollBar Component", () => {
    const tree = create(
      <ScrollBar title="ScrollBar Title">
        <h1>Employee</h1>
      </ScrollBar>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});