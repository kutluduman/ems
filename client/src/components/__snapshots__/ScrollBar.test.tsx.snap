// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollBar Component renders the ScrollBar Component 1`] = `
<div
  className="css-jjtu05"
>
  <div
    className="css-wkgdtb"
    data-simplebar={true}
    title="ScrollBar Title"
  >
    <div
      className="simplebar-wrapper"
    >
      <div
        className="simplebar-height-auto-observer-wrapper"
      >
        <div
          className="simplebar-height-auto-observer"
        />
      </div>
      <div
        className="simplebar-mask"
      >
        <div
          className="simplebar-offset"
        >
          <div
            className="simplebar-content-wrapper"
          >
            <div
              className="simplebar-content"
            >
              <h1>
                Employee
              </h1>
            </div>
          </div>
        </div>
      </div>
      <div
        className="simplebar-placeholder"
      />
    </div>
    <div
      className="simplebar-track simplebar-horizontal"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
    <div
      className="simplebar-track simplebar-vertical"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
  </div>
</div>
`;
